### PR TITLE
Replace `send_onchain` snippets with `pay_onchain`

### PR DESCRIPTION
**Pull request by ok300**
This PR replaces the old `send_onchain` call in snippets with `pay_onchain`.

### CODE REVIEW

File: snippets/go/send_onchain.go [hunk 1]
```diff
@@ -26,29 +26,25 @@ func PreparePayOnchain(currentLimits breez_sdk.OnchainPaymentLimitsResponse) {
...
-			SatPerVbyte:             satPerVbyte,
-		}
-		if reverseSwapInfo, err := sdk.SendOnchain(sendOnchainRequest); err == nil {
-			log.Printf("%#v", reverseSwapInfo)
-		}
+
+	payOnchainRequest := breez_sdk.PayOnchainRequest{
+		RecipientAddress:   destinationAddress,
+		PrepareRes:         prepareRes,
+	}
```
Comment (dangeross) [inline snippets/go/send_onchain.go:44]: Fix CI
```suggestion
	payOnchainRequest := breez_sdk.PayOnchainRequest{
		RecipientAddress: destinationAddress,
		PrepareRes:       prepareRes,
	}
```

File: snippets/react-native/send_onchain.ts
Comment (dangeross) [inline snippets/react-native/send_onchain.ts:1]: Need to add the imports for the CI to pass:
```
import {
  type OnchainPaymentLimitsResponse,
  type PrepareOnchainPaymentResponse,
  fetchReverseSwapFees,
  inProgressReverseSwaps,
  onchainPaymentLimits,
  payOnchain,
  prepareOnchainPayment,
  SwapAmountType,
  maxReverseSwapAmount
} from '@breeztech/react-native-breez-sdk'
```

File: snippets/dart_snippets/lib/send_onchain.dart [hunk 2]
```diff
@@ -20,28 +20,24 @@ Future<PrepareOnchainPaymentResponse> preparePayOnchain({
...
-    pairHash: pairHash,
-    satPerVbyte: satPerVbyte,
+  PayOnchainRequest req = PayOnchainRequest(
+    recipientAddress: onchainRecipientAddress,
+    prepareRes: prepareRes,
   );
-  SendOnchainResponse resp = await BreezSDK().sendOnchain(req: req);
+  PayOnchainResponse res = await BreezSDK().payOnchain(req: req);
   // ANCHOR_END: start-reverse-swap
   return resp;
```
Comment (dangeross) [inline snippets/dart_snippets/lib/send_onchain.dart:42]: To fix CI
```suggestion
  return res;
```

File: snippets/dart_snippets/lib/send_onchain.dart [hunk 3]
```diff
@@ -20,28 +20,24 @@ Future<PrepareOnchainPaymentResponse> preparePayOnchain({
...
-  PrepareOnchainPaymentResponse resp = await BreezSDK().prepareOnchainPayment(req: req);
-  print("Sender amount: ${resp.senderAmountSat} sats");
-  print("Recipient amount: ${resp.recipientAmountSat} sats");
-  print("Total fees: ${resp.totalFees} sats");
+  PrepareOnchainPaymentResponse prepareRes = await BreezSDK().prepareOnchainPayment(req: req);
+  print("Sender amount: ${prepareRes.senderAmountSat} sats");
+  print("Recipient amount: ${prepareRes.recipientAmountSat} sats");
+  print("Total fees: ${prepareRes.totalFees} sats");
   // ANCHOR_END: prepare-pay-onchain
   return resp;
```
Comment (dangeross) [inline snippets/dart_snippets/lib/send_onchain.dart:28]:
```suggestion
  return prepareRes;
```

File: snippets/react-native/send_onchain.ts [hunk 4]
```diff
@@ -43,18 +43,14 @@ const examplePreparePayOnchain = async (currentLimits: OnchainPaymentLimitsRespo
   // ANCHOR_END: prepare-pay-onchain
 }
 
-const exampleSendOnchain = async (currentFees: ReverseSwapPairInfo) => {
+const examplePayOnchain = async (prepareResponse: PrepareOnchainPaymentResponse) => {
```
Comment (dangeross) [inline snippets/react-native/send_onchain.ts:46]:
```suggestion
const examplePayOnchain = async (prepareRes: PrepareOnchainPaymentResponse) => {
```

File: snippets/react-native/send_onchain.ts [hunk 5]
```diff
@@ -43,18 +43,14 @@ const examplePreparePayOnchain = async (currentLimits: OnchainPaymentLimitsRespo
...
-    const satPerVbyte = 5
 
-    const reverseSwapInfo = await sendOnchain({
-      amountSat,
-      onchainRecipientAddress,
-      pairHash: currentFees.feesHash,
-      satPerVbyte
+    const reverseSwapInfo = await payOnchain({
+      recipientAddress: onchainRecipientAddress,
+      prepareResponse
```
Comment (dangeross) [inline snippets/react-native/send_onchain.ts:53]:
```suggestion
      prepareRes
```

**Review by dangeross — CHANGES_REQUESTED** [review]

File: snippets/python/src/send_onchain.py
Comment (dangeross) [inline snippets/python/src/send_onchain.py:1]: The return variable needs changing to `current_limits` for get_current_limits:
```
def get_current_limits(sdk_services):
    try: 
        # ANCHOR: get-current-reverse-swap-limits
        current_limits = sdk_services.onchain_payment_limits()
        print("Minimum amount, in sats: ", current_limits.min_sat)
        print("Maximum amount, in sats: ", current_limits.max_sat)
        # ANCHOR_END: get-current-reverse-swap-limits
        return current_limits
    except Exception as error:
        print(error)
        raise
```

File: snippets/dart_snippets/lib/send_onchain.dart
Comment (erdemyerebasmaz) [inline snippets/dart_snippets/lib/send_onchain.dart:1]: Could rename this file to `pay_onchain.dart`

File: snippets/python/src/send_onchain.py
Comment (ok300) [reply snippets/python/src/send_onchain.py:1]: Done: c6e014b6cb12dfa9975937bd16bc458fd1fd1a2d

File: snippets/react-native/send_onchain.ts
Comment (ok300) [reply snippets/react-native/send_onchain.ts:1]: Done: 3df4fc04d85a5ad96aca3e8c834861b644ed47e7

File: snippets/dart_snippets/lib/send_onchain.dart
Comment (ok300) [reply snippets/dart_snippets/lib/send_onchain.dart:1]: Good idea. I added it as a separate step in https://github.com/breez/breez-sdk-docs/issues/140 to make it easier to review.

**Comment by ok300** [comment]
@dangeross @erdemyerebasmaz thanks for the findings, fixed them all. Ready for final review.

**Review by dangeross — APPROVED** [review]
LGTM

**Review by erdemyerebasmaz — APPROVED** [review]
LGTM

**Pull request merged**